src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/DiseaseController.kt

```diff
@@ -0,0 +1,33 @@
+package org.imperial.mrc.hint.controllers
+
+import org.imperial.mrc.hint.APIClient
+import org.imperial.mrc.hint.FileManager
+import org.imperial.mrc.hint.FileType
+import org.imperial.mrc.hint.asResponseEntity
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+import org.springframework.web.multipart.MultipartFile
+
+@RestController
+@RequestMapping("/disease")
+class DiseaseController(private val fileManager: FileManager,
+                        private val apiClient: APIClient) {
+
+    @PostMapping("/survey/")
+    @ResponseBody
+    fun uploadSurvey(@RequestParam("file") file: MultipartFile): ResponseEntity<String> {
+
+        val path = fileManager.saveFile(file, FileType.Survey)
```

Comment:
Could pull these three lines out into a saveAndValidate(file, fileType) method, maybe in a FileValidationController base class for DiseaseController and BaselineController?

Reply:
Cool, have pulled out into private method and can pull out base class for the controllers when hintr is in a more usable state and the baseline methods get wired up properly.